test_runner/fixtures/neon_fixtures.py

```diff
@@ -2817,8 +2817,15 @@ def check_restored_datadir_content(
     endpoint: Endpoint,
 ):
     # Get the timeline ID. We need it for the 'basebackup' command
-    timeline = TimelineId(endpoint.safe_psql("SHOW neon.timeline_id")[0][0])
+    timeline_id = TimelineId(endpoint.safe_psql("SHOW neon.timeline_id")[0][0])
 
+    # many tests already checkpoint, but do it just in case
```

Comment:
Let's just remove it from all tests and leave it here?

Reply:
Callsites of this function:

`test_runner/regress/test_multixact.py::test_multixact` -- I don't think this can be removed, because it will probably advance the lsn, which is then used to branch and so on. You commented in chat similar I think.

`test_runner/regress/test_pg_regress.py::test_sql_regress` -- what side effects does `pg_current_wal_insert_lsn` have? Should we do it after the `CHECKPOINT` in the check function?

`test_runner/regress/test_createdropdb.py::test_dropdb` does not have a `CHECKPOINT` at all...?

Easy to remove `CHECKPOINT`:

- `test_runner/regress/test_pg_regress.py::test_pg_regress` -- easy
- `test_runner/regress/test_subxacts.py::test_subxacts` -- easy

Reply:
> test_runner/regress/test_multixact.py::test_multixact -- I don't think this can be removed, because it will probably advance the lsn

I think this doesn't matter, but not too important, can leave as well.

 > what side effects does pg_current_wal_insert_lsn have?

Nothing, it looks like debugging accidental leftover.

Reply:
Waiting for the next test run on [65e06ab](https://github.com/neondatabase/neon/pull/4903/commits/65e06abfcb0abb9a52e16a1e2578f11e1a7a6e6d) and pushing the cleanups for other than `test_multixact`.

Reply:
[`test_s3_wal_replay[debug-pg15-local_fs]`](https://neon-github-public-dev.s3.amazonaws.com/reports/pr-4903/5820335253/index.html#suites/82004ab4e3720b47bf78f312dabe7c55/f347621f5ddd6692) was the only one flaky for 65e06ab, which is great, but these failures have been quite rare.

0782902b9 will have the cleanup.

Reply:
0 test failures, assuming https://github.com/neondatabase/neon/commit/0782902b98dad50781a16b24696eb0c3f94a409e was good.